src/modules/map/components/CompareSlider.vue

```diff
@@ -1,53 +1,109 @@
 <script setup>
 import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
-import { computed, defineProps, onMounted, ref, toRefs, watch } from 'vue'
+import { unByKey } from 'ol/Observable'
+import { computed, inject, nextTick, onMounted, ref, watch } from 'vue'
 import { useStore } from 'vuex'
 
 import { round } from '@/utils/numberUtils'
-const props = defineProps({
-    clientWidth: {
-        type: Number,
-        default: window.innerWidth,
-    },
-})
+const getMap = inject('getMap')
 
-const { clientWidth } = toRefs(props)
+const preRenderKey = ref(null)
+const postRenderKey = ref(null)
+const compareSliderOffset = ref(0)
 
 const compareRatio = ref(-0.5)
-
 const store = useStore()
 const storeCompareRatio = computed(() => store.state.ui.compareRatio)
+const clientWidth = computed(() => store.state.ui.width)
 const compareSliderPosition = computed(() => {
     return {
         left: compareRatio.value * 100 + '%',
     }
 })
+const visibleLayerOnTop = computed(() => store.getters.visibleLayerOnTop)
 
 watch(storeCompareRatio, (newValue) => {
     compareRatio.value = newValue
+    slice()
+})
+
+watch(visibleLayerOnTop, () => {
+    nextTick(slice())
```

Comment:
Not sure this works as intended if you call the slice instead of passing it's reference.
You maybe should change that to either `nextTick(slice)` or `nextTick(() => slice())`

Reply:
it seemed to work, but it works just as well with `nextTick(slice)`, so might as well use it the intended way.